internal/resolver/resolve.go

```diff
@@ -280,11 +354,11 @@ func (p *parser) resolveVars(prog *Program) {
 	// assign indexes basically randomly)
 	prog.Scalars = make(map[string]int)
 	prog.Arrays = make(map[string]int)
-	for name, info := range p.varTypes[""] {
-		_, isFunc := p.functions[name]
+	for name, info := range r.varTypes[""] {
+		_, isFunc := r.functions[name]
 		if isFunc {
 			// Global var can't also be the name of a function
-			panic(p.errorf("global var %q can't also be a function", name))
+			panic(prog.EndPos.Errorf("global var %q can't also be a function", name))
```

Comment:
This is only the program's "end pos" because we don't have access to a better position at this point. Ideally it'd be the position of the global var. For now, I think we should just make it position 1,1 (that's at last as good as the program's end pos) and remove `Program.EndPos`.